lib/compact.js

```diff
@@ -6,8 +6,36 @@
  * @returns {Array}
  */
 
+function isFalsy(val) {
+
+  if (val == null || val == undefined) {
+    return true;
+  }
+
+  let first = (val == false || val == 0 || val == "");
+  let second = isNaN(val) && val.constructor == Number;
+
+  if (val.constructor != Number) {
+    return first;
+  }
+
+  return first || second;
+}
+
 function compact(arr) {
-  throw new Error("This method is still a prototype");
+
+  if (arr == null) {
+    return new Array();
+  }
+
+  var newArray = new Array();
+  arr.forEach(function(val) {
+    if (isFalsy(val) == false){
+      newArray.push(val);
+    }
+  });
+
+  return newArray;
```

Comment:
We really appreciate your contribution and look forward to getting this merged but I think we should look at the implementation.  Compact can be a really simple function with the right tools. Look into `Array.prototype.filter` and the identity function. I think you'll be surprised.